scm/mock.go

```diff
@@ -23,7 +22,9 @@ type MockSCM struct {
 }
 
 // NewMockSCMClient returns a new mock client implementing the SCM interface.
-func NewMockSCMClient() *MockSCM {
+// If withCourse is true, creates default course repositories and teams for a course
+// associated with qtest.MockOrg mock organization.
+func NewMockSCMClient(withCourse bool) *MockSCM {
```

Comment:
I don’t like passing bools like this. It has wide-ranging consequences (lots of unrelated changes in other code). 

It is better to add another constructor `NewMockSCMClientWithCourse`… and similar for other mocks that also require with course.